### PR TITLE
Use a build flag to disable resumable digests

### LINKED ISSUE #341: Make the use of resumable digest optional
There are concerns about the use of a forked version of sha (see #295 for context).

Possible solution: have a configuration variable making it possible to toggle it on/off.

### PR DESCRIPTION
**Pull request by ncdc**
Signed-off-by: Andy Goldstein <agoldste@redhat.com>

Closes #341.

### CODE REVIEW

**Comment by vbatts** [comment]
oh cool. I thought @stevvooe eluded to this already being in here, but had not checked yet. 
Will review.

**Comment by ncdc** [comment]
Thanks, this was a quick stab at it, so I'd definitely appreciate feedback.

File: digest/digester_resumable.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// +build resumabledigest
```
Comment (stevvooe) [inline digest/digester_resumable.go]: Let's call the flag "noresumabledigest", so default is to build it with resumable.

**Comment by stevvooe** [comment]
@ncdc Per our IRC conversation, let's see if we can avoid forking `layerwriter.go`

**Comment by stevvooe** [comment]
LGTM!

cc @jlhawn @dmcgowan

**Comment by dmp42** [comment]
LGTM

**Pull request merged**